flink-ml-lib/src/main/java/org/apache/flink/ml/feature/onehotencoder/OneHotEncoderModel.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+import org.apache.flink.table.runtime.typeutils.ExternalTypeInfo;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Vector;
+import java.util.function.Function;
+
+/**
+ * A Model which encodes data into one-hot format using the model data computed by {@link
+ * OneHotEncoder}.
+ */
+public class OneHotEncoderModel
+        implements Model<OneHotEncoderModel>, OneHotEncoderParams<OneHotEncoderModel> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+    private Table modelTable;
+
+    public OneHotEncoderModel() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        final String[] inputCols = getInputCols();
+        final String[] outputCols = getOutputCols();
+        final boolean dropLast = getDropLast();
+        final String broadcastModelKey = "OneHotModelStream";
+
+        Preconditions.checkArgument(getHandleInvalid().equals("ERROR"));
+        Preconditions.checkArgument(inputs.length == 1);
+        Preconditions.checkArgument(inputCols.length == outputCols.length);
+
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                Collections.nCopies(
+                                                outputCols.length,
+                                                ExternalTypeInfo.of(Vector.class))
+                                        .toArray(new TypeInformation[0])),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), outputCols));
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelTable).getTableEnvironment();
+        DataStream<Row> input = tEnv.toDataStream(inputs[0]);
+        DataStream<Tuple2<Integer, Integer>> modelStream =
+                OneHotEncoderModelData.getModelDataStream(tEnv, modelTable);
+
+        Map<String, DataStream<?>> broadcastMap = new HashMap<>();
+        broadcastMap.put(broadcastModelKey, modelStream);
+
+        Function<List<DataStream<?>>, DataStream<Row>> function =
+                dataStreams -> {
+                    DataStream stream = dataStreams.get(0);
+                    return stream.transform(
+                            PredictLabelOperator.class.getSimpleName(),
```

Comment:
nits: could we just use "PredictLabelOperator" for simplicity here?

Reply:
OK. I'll make the change.